packages/central-server/src/apiV2/utilities/SurveyResponseImporter.js

```diff
@@ -46,9 +50,13 @@ export class SurveyResponseImporter {
     const survey = await this.models.survey.findOne({ name: surveyName });
     const responses = await this.getResponsesForSurvey(survey, rows);
 
-    return this.models.wrapInTransaction(async transactingModels =>
-      submitResponses(transactingModels, userId, responses),
-    );
+    return this.models.wrapInTransaction(async transactingModels => {
+      // Upsert entities and options that were created in user's local database
+      await upsertEntitiesAndOptions(transactingModels, responses);
```

Comment:
Just checking, we don't need to check permissions here like in the `SubmitSurveyResponses` route?

Reply:
We check the permissions for using the importer before we begin the import. Surprisingly we don't use the `SurveyResponseImporter` in `importSurveyResponses` :sweat_smile: but we do use it in `importStriveLabResults` and we check the permissions first there :+1: